ui/apps/platform/src/Containers/MainPage/Navigation/HorizontalSubnav.tsx

```diff
@@ -0,0 +1,234 @@
+                      path: vulnerabilitiesNodeCvesPath,
+                      routeKey: 'vulnerabilities/node-cves',
+                  },
+                  {
+                      type: 'parent',
+                      key: 'More Views',
+                      title: 'More Views',
+                      children: [
+                          {
+                              type: 'link',
+                              content: 'All Images',
+                              description:
+                                  'View findings for user and platform images simultaneously',
+                              // TODO Change these
+                              path: '/main/vulnerabilities/TBD',
+                              routeKey: 'vulnerabilities/workload-cves',
+                          },
+                      ],
+                  },
+              ]
+            : [],
+    };
+}
+
+/*
+ * Given the mapping of parent routes to subnav description groups, return the grouping
+ * that contains a child item with a path matching the user's current path
+ */
+function getSubnavGroupForCurrentPath(
+    subnavDescriptionGroups: Record<SubnavParentKey, NavDescription[]>,
+    pathname: string
+) {
+    return (
+        Object.values(subnavDescriptionGroups).find((subnavDescriptionGroup) => {
+            return subnavDescriptionGroup.some((group) => {
+                if (group.type === 'link') {
+                    return matchPath(pathname, group.path);
+                }
+                return group.children
+                    .filter((child) => child.type === 'link')
+                    .some(({ path }) => matchPath(pathname, path));
+            });
+        }) ?? []
+    );
+}
+
+export type HorizontalSubnavProps = {
+    hasReadAccess: HasReadAccess;
+    isFeatureFlagEnabled: IsFeatureFlagEnabled;
+};
+
+function HorizontalSubnav({ hasReadAccess, isFeatureFlagEnabled }: HorizontalSubnavProps) {
+    const history = useHistory();
+    const { pathname } = useLocation();
+    const routePredicates = { hasReadAccess, isFeatureFlagEnabled };
+
+    const subnavDescriptionGroups = getSubnavDescriptionGroups(isFeatureFlagEnabled);
+    const subnavDescriptionGroupForCurrentPath = getSubnavGroupForCurrentPath(
+        subnavDescriptionGroups,
+        pathname
+    );
+    const subnavDescriptions = filterNavDescriptions(
+        subnavDescriptionGroupForCurrentPath,
+        routePredicates
+    );
+
+    const [openDropdownKey, setOpenDropdownKey] = useState<string | null>(null);
+
+    const onToggleClick = (key: string) => {
+        setOpenDropdownKey((currentKey) => (currentKey === key ? null : key));
+    };
+
+    const onSelect = (
+        _event: React.MouseEvent<Element, MouseEvent> | undefined,
+        value: string | number | undefined
+    ) => {
+        history.push(value);
+        setOpenDropdownKey(null);
+    };
+
+    if (!subnavDescriptions.length) {
+        return null;
+    }
```

Comment:
Question to refresh my understanding, not necessarily a change request.

What are pro and con if early exit immediately follows `subnavDescriptions` assignment and therefore precedes callback functions that are called only for render of `Nav` element?
* On one side of coin, TypeScript does not need to understand whether or not they can be called.
* On same side of coin, `useState` hook must precede any conditional code, so it would split that chunk of related content.

Reply:
I think the main point in favor of moving the early return would be readability. i.e. None of the stuff below this guard condition matters if the condition is hit.

The downside is as you mentioned, the `useState` cannot be called conditionally. I tend to prefer the approach you mentioned, keeping early return guard conditions as close to the top as possible, except in React components due to hooks interfering with the ideal ordering too often. We could extract the `subnavDescriptions` logic to above the component, but I'm not sure if that would aid or hamper readability.